### PR TITLE
records: add schema.org json-ld serializer

### LINKED ISSUE #2273: records: add schema.org serialiser
Introduce schema.org serialiser that outputs in JSON-LD.  (Stemmed from #1324.)

We can start with simple mappings (Datasets, Software) and leave harder mappings (configuration files? Other?) for later. 

See also:

- cernopendata's `DataCiteSerializer` 
- `zenodo/modules/records/serializers/schemaorg.py`

### LINKED ISSUE #2275: records: inject JSON-LD
Once we have schema.org serialiser (#2273), enrich detailed record pages by injecting JSON-LD with it there.

Consider (1) injecting full content directly to `/record/NNN` page body vs (2) injecting only reference to other URL exposing JSON-LD such as `/record/NNN/export/FFF`.

Start with simple metadata author/title/type etc for the first prototype, but think of the later use case about how to deal with numerous file assets.  (Some datasets can have more than 10,000 files to download, so some sort of pagination would be nice to have perhaps.)

### PR DESCRIPTION
**Pull request by hjhsalo**
* Add schema.org JSON-LD serializer
  (closes #2273)
* Add schema.org JSON-LD description of Record to detail page of
  Dataset and Software Records.
  (closes #2275)

Signed-off-by: Harri Hirvonsalo <harri.hirvonsalo@cern.ch>

### CODE REVIEW

**Comment by hjhsalo** [comment]
Test for example with `curl` or check sources of rendered detail page of a record.

```
$ curl -H "Accept: application/ld+json" http://127.0.0.1:5000/api/records/221
{
  "type": "Software"
}
```

```
$ curl -H "Accept: application/ld+json" http://127.0.0.1:5000/api/records/1100
{
  "type": "Dataset"
}
```

```
$ curl http://127.0.0.1:5000/record/1100 | grep -A 4 "<script type='application/ld+json'>"
<script type='application/ld+json'>
{
  "type": "Dataset"
}
</script>
```

```
$ curl http://127.0.0.1:5000/record/221 | grep -A 4 "<script type='application/ld+json'>"
<script type='application/ld+json'>
{
  "type": "Software"
}
</script>
```

File: cernopendata/modules/records/serializers/schemaorg.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+class OpendataSchemaorgSerializer(JSONSerializer):
+    """Schema for records v1 in JSON.
+
+    Serializes a Record based on it's type (Dataset, Software, etc.) to
+    schema.org compatible JSON-LD syntax.
+
+    """
+
+    def preprocess_record(self, pid, record, links_factory=None):
```
Comment (ioannistsanaktsidis) [inline cernopendata/modules/records/serializers/schemaorg.py]: This one can be skipped completely if it returns only result, as dump is doing the job.

File: cernopendata/modules/records/serializers/schemas/schemaorg.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class DatasetSchemaorgSchema(Schema):
+    """schema.org JSON-LD schema for CERN Opendata Dataset Record."""
+
+    # TODO: Implement schema.org structure and keys
+
+    type = fields.Method('get_type')
+
+    def get_type(self, obj):
```
Comment (ioannistsanaktsidis) [inline cernopendata/modules/records/serializers/schemas/schemaorg.py]: I don't really get the use of get_type as here https://github.com/cernopendata/opendata.cern.ch/pull/2277/files#diff-88d5e62a55ecff4c110e58decd763ce6R55 you check the type directly.

Comment (hjhsalo) [reply cernopendata/modules/records/serializers/schemas/schemaorg.py]: Sorry, I force pushed new commit before I had a look.

**Comment by hjhsalo** [comment]
Closing this in favor of #2295
(this PR is made against master-branch, while it should have been made against qa-branch)

**Pull request closed without merging**